dataprofiler/data_readers/json_data.py

```diff
@@ -51,83 +56,201 @@ def __init__(self, input_file_path=None, data=None, options=None):
         #  _selected_data_format: user selected format in which to return data
         #                         can only be of types in _data_formats
         #  _selected_keys: keys being selected from the entire dataset
+        #  _payload_keys: dictionary key that determines the payload
+
         self._data_formats["records"] = self._get_data_as_records
         self._data_formats["json"] = self._get_data_as_json
-        self._selected_data_format = options.get("data_format", "dataframe")
+        self._data_formats["flattened_dataframe"] = self._get_data_as_flattened_dataframe
+        self._selected_data_format = options.get("data_format", "flattened_dataframe")
+        self._payload_keys = options.get("payload_keys", ["data", "payload", "response"])
+        if not isinstance(self._payload_keys, list):
```

Comment:
Should you make sure its a string first or can it be float/int too?

Reply:
Technically json only has string keys, but dictionaries don't need to be. So maybe the data is sent in as a dict to be read by JSON data with integer keys or maybe someone will send in an integer key but actually mean string. I'm not sure which use case would ever happen so I'm not sure what to do here, but I dont think it really matters